ml-algorithms/src/main/java/org/opensearch/ml/engine/tools/SearchIndexTool.java

```diff
@@ -56,12 +56,22 @@ public class SearchIndexTool implements Tool {
     private static final String DEFAULT_DESCRIPTION =
         "Use this tool to search an index by providing two parameters: 'index' for the index name, and 'query' for the OpenSearch DSL formatted query. Only use this tool when both index name and DSL query is available.";
 
-    public static final String DEFAULT_INPUT_SCHEMA = "{\"type\":\"object\","
-        + "\"properties\":{\"index\":{\"type\":\"string\",\"description\":\"OpenSearch index name. for example: index1\"},"
-        + "\"query\":{\"type\":\"object\",\"description\":\"OpenSearch search index query. You need to get index mapping to write correct search query. It must be a valid OpenSearch query."
-        + " Valid value:\\n{\\\"query\\\":{\\\"match\\\":{\\\"population_description\\\":\\\"seattle 2023 population\\\"}},\\\"size\\\":2,\\\"_source\\\":\\\"population_description\\\"}\\n"
-        + "Invalid value: \\n{\\\"match\\\":{\\\"population_description\\\":\\\"seattle 2023 population\\\"}}\\nThe value is invalid because the match not wrapped by \\\"query\\\".\","
-        + "\"additionalProperties\":false}},\"required\":[\"index\",\"query\"],\"additionalProperties\":false}";
+    public static final String DEFAULT_INPUT_SCHEMA =
+        """
+            {
+              "input": {
+                "index": {
+                  "type": "string",
+                  "description": "OpenSearch index name. for example: index1"
+                },
+                "query": {
+                  "type": "object",
+                  "description": "OpenSearch search index query. You need to get index mapping to write correct search query. It must be a valid OpenSearch query. Valid value:\\n{\\"query\\":{\\"match\\":{\\"population_description\\":\\"seattle 2023 population\\"}},\\"size\\":2,\\"_source\\":\\"population_description\\"}\\nInvalid value: \\n{\\"match\\":{\\"population_description\\":\\"seattle 2023 population\\"}}\\nThe value is invalid because the match not wrapped by \\"query\\".",
+                  "additionalProperties": false
```

Comment:
This part is not a direct string reformat. I see `\"required\":[\"index\",\"query\"]` was removed.

Can you explain why? To keep this PR scope clear, I would suggest put such reformatting/refactoring in a new PR

Reply:
reverted.